annotations/transport/annotations_router.go

```diff
@@ -0,0 +1,236 @@
+		h.api.Err(w, r, err)
+		return
+	}
+
+	h.api.Respond(w, r, http.StatusNoContent, nil)
+}
+
+func (h *AnnotationHandler) handleGetAnnotation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	id, err := platform.IDFromString(chi.URLParam(r, "id"))
+	if err != nil {
+		h.api.Err(w, r, errBadAnnotationId)
+		return
+	}
+
+	a, err := h.annotationService.GetAnnotation(ctx, *id)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	h.api.Respond(w, r, http.StatusOK, a)
+}
+
+func (h *AnnotationHandler) handleDeleteAnnotation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	id, err := platform.IDFromString(chi.URLParam(r, "id"))
+	if err != nil {
+		h.api.Err(w, r, errBadAnnotationId)
+		return
+	}
+
+	if err := h.annotationService.DeleteAnnotation(ctx, *id); err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	h.api.Respond(w, r, http.StatusNoContent, nil)
+}
+
+func (h *AnnotationHandler) handleUpdateAnnotation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	id, err := platform.IDFromString(chi.URLParam(r, "id"))
+	if err != nil {
+		h.api.Err(w, r, errBadAnnotationId)
+		return
+	}
+
+	u, err := decodeUpdateAnnotationRequest(r)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	a, err := h.annotationService.UpdateAnnotation(ctx, *id, *u)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	h.api.Respond(w, r, http.StatusOK, a)
+}
+
+func decodeCreateAnnotationsRequest(r *http.Request) ([]influxdb.AnnotationCreate, error) {
+	cs := []influxdb.AnnotationCreate{}
+	if err := json.NewDecoder(r.Body).Decode(&cs); err != nil {
+		return nil, err
+	}
+
+	for _, c := range cs {
+		if err := c.Validate(time.Now); err != nil {
+			return nil, err
+		}
+	}
+
+	return cs, nil
+}
+
+func decodeListAnnotationsRequest(r *http.Request) (*influxdb.AnnotationListFilter, error) {
+	startTime, endTime, err := tFromReq(r)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &influxdb.AnnotationListFilter{
+		StreamIncludes: r.URL.Query()["streamIncludes"],
+		BasicFilter: influxdb.BasicFilter{
+			EndTime:   endTime,
+			StartTime: startTime,
+		},
+	}
+	f.SetStickerIncludes(r.URL.Query())
+	if err := f.Validate(time.Now); err != nil {
+		return nil, err
+	}
+
+	return f, nil
+}
+
+func decodeDeleteAnnotationsRequest(r *http.Request) (*influxdb.AnnotationDeleteFilter, error) {
+	// Try to get a stream ID from the query params. The stream ID is not required,
+	// so if one is not set we can leave streamID as the zero value.
+	var streamID platform.ID
+	if qid := chi.URLParam(r, "streamID"); qid != "" {
+		id, err := platform.IDFromString(qid)
+		// if a streamID parameter was provided but is not valid, return an error
+		if err != nil {
+			return nil, errBadStreamId
+		}
+		streamID = *id
+	}
+
+	startTime, endTime, err := tFromReq(r)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &influxdb.AnnotationDeleteFilter{
+		StreamTag: r.URL.Query().Get("stream"),
+		StreamID:  streamID,
```

Comment:
Should we worry/guard against users specifying both `streamID` and `stream`, and having the 2 values disagree with each other?

Reply:
That's a good point and probably a good idea, but I don't want to introduce inconsistencies between the cloud and OSS API, since the cloud API allows for both to be specified, and gives precedence to a `streamID` (in the to-be-implemented persistence code). Might be something to think about down the road to change in both places, particularly if we make the API available outside of the UI.